src/components/image/image.tsx

```diff
@@ -70,30 +81,48 @@ export class Image extends React.PureComponent<ImageProps, ImageState> {
                 src={this.state.src}
                 onLoad={this.onLoad}
                 onError={this.onError}
+                style-state={styleState}
             />
         );
     }
     public componentWillMount() {
-        this.setState({src: this.props.src || this.props.defaultImage!});
+        this.setState({
+            src: this.props.src || this.props.defaultImage!,
+            status: ImageStatus.Loading
+        });
     }
 
     public componentWillReceiveProps(newProps: ImageProps) {
-        this.setState({src: newProps.src || this.props.defaultImage!});
+        this.setState({
+            src: newProps.src || this.props.defaultImage!,
+            status: ImageStatus.Loading
+        });
     }
 
     private onError: React.EventHandler<React.SyntheticEvent<HTMLImageElement>> = e => {
+        this.setState({
+            src: this.getFallbackSrcFor(this.state.src),
+            status: ImageStatus.Error
+        });
         this.props.onError!({...e, src: this.state.src});
-        this.setState({src: this.getFallbackSrcFor(this.state.src)});
     }
 
     private onLoad: React.EventHandler<React.SyntheticEvent<HTMLImageElement>> = e => {
-        if (this.state.src !== this.props.defaultImage && this.state.src !== transparentImage) {
+        if (this.state.src !== this.props.defaultImage &&
+            this.state.src !== transparentImage &&
+            this.state.src !== this.props.errorImage) {
+
+            this.setState({status: ImageStatus.Loaded});
             this.props.onLoad!({...e, src: this.state.src});
         }
     }
 
     private getFallbackSrcFor(src: string): string {
-        // first, fallback to defaultImage, and later to one transparent pixel
-        return (src !== this.props.defaultImage) ? this.props.defaultImage! : transparentImage;
+        // first, fallback to errorImage, and later to one transparent pixel
+        if (!this.props.errorImage) {
```

Comment:
why a separate if? you could add this condition to the ternary expression:
````
const { errorImage } = this.props;
return !errorImage || src === errorImage ? transparentImage : errorImage;
```

Reply:
This is a matter of code style. I have no problem with your suggestion but at different times got comments to take apart such single line multiple conditionals.